job/job.go

```diff
@@ -190,14 +193,28 @@ func (j *Job) UnmarshalJSON(b []byte) error {
 	}
 	j.DownloadTimeout = timeout
 
-	var useragent string
-	if useragentField, ok := tmp["user_agent"]; ok {
-		useragent, ok = useragentField.(string)
-		if !ok {
-			return errors.New("UserAgent must be a string")
+	rh, ok := tmp["request_headers"]
+	if ok {
+		if requestHeaders, ok := rh.(map[string]interface{}); ok {
+			headers := make(map[string]string)
+			for k, v := range requestHeaders {
+				switch v.(type) {
+				case string:
+					headers[k] = v.(string)
+				case int:
+					headers[k] = strconv.Itoa(v.(int))
+				default:
+					// According to https://tools.ietf.org/html/rfc2616#section-5.3
```

Comment:
I am not sure that we should ignore these values. Clients might mistakenly "believe" that their headers are accepted although they are silently ignored.

Regarding accepted types, we could probably get away with only supporting strings and leaving unmarshaling to Go.

Reply:
In the `default` we return an error.